internal/provider/resource_gitlab_service_external_wiki_test.go

```diff
@@ -0,0 +1,125 @@
+	var externalWikiURL2 = "http://mynumbertwowiki.com"
+	var externalWikiResourceName = "gitlab_service_external_wiki.this"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabServiceExternalWikiDestroy,
+		Steps: []resource.TestStep{
+			// Create an External Wiki service
+			{
+				Config: testAccGitlabServiceExternalWikiConfig(testProject.ID, externalWikiURL1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabServiceExternalWikiExists(externalWikiResourceName, &externalWikiService),
+					resource.TestCheckResourceAttr(externalWikiResourceName, "external_wiki_url", externalWikiURL1),
+				),
+			},
+			// Verify import
+			{
+				ResourceName:      "gitlab_service_external_wiki.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// Update the External Wiki service
+			{
+				Config: testAccGitlabServiceExternalWikiConfig(testProject.ID, externalWikiURL2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabServiceExternalWikiExists(externalWikiResourceName, &externalWikiService),
+					resource.TestCheckResourceAttr(externalWikiResourceName, "external_wiki_url", externalWikiURL2),
+				),
+			},
+			// Verify import
+			{
+				ResourceName:      "gitlab_service_external_wiki.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// Update the External Wiki service to get back to previous settings
+			{
+				Config: testAccGitlabServiceExternalWikiConfig(testProject.ID, externalWikiURL1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabServiceExternalWikiExists(externalWikiResourceName, &externalWikiService),
+					resource.TestCheckResourceAttr(externalWikiResourceName, "external_wiki_url", externalWikiURL1),
+				),
+			},
+			// Verify import
+			{
+				ResourceName:      "gitlab_service_external_wiki.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckGitlabServiceExternalWikiExists(resourceIdentifier string, service *gitlab.ExternalWikiService) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceIdentifier]
+		if !ok {
+			return fmt.Errorf("Not Found: %s", resourceIdentifier)
+		}
+
+		project := rs.Primary.Attributes["project"]
+		if project == "" {
+			return fmt.Errorf("No project ID is set")
+		}
+
+		externalWikiService, _, err := testGitlabClient.Services.GetExternalWikiService(project)
+		if err != nil {
+			return fmt.Errorf("External Wiki service does not exist in project %s: %v", project, err)
+		}
+		*service = *externalWikiService
+
+		return nil
+	}
+}
+
+func testAccCheckGitlabServiceExternalWikiDestroy(s *terraform.State) error {
+	var project string
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type == "gitlab_project_environment" {
```

Comment:
This looks like a copy-paste-bug

Reply:
Gosh darned it. I copied part of the logic and forgot to change the resource 🥇

Reply:
All fixed